include/alpaka/dev/DevOacc.hpp

```diff
@@ -0,0 +1,341 @@
+/* Copyright 2019 Benjamin Worpitz
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#ifdef ALPAKA_ACC_ANY_BT_OACC_ENABLED
+
+#if _OPENACC < 201306
+    #error If ALPAKA_ACC_ANY_BT_OACC_ENABLED is set, the compiler has to support OpenACC xx or higher!
+#endif
+
+#include <alpaka/dev/Traits.hpp>
+#include <alpaka/mem/buf/Traits.hpp>
+#include <alpaka/pltf/Traits.hpp>
+#include <alpaka/wait/Traits.hpp>
+
+#include <alpaka/queue/Traits.hpp>
+#include <alpaka/queue/Properties.hpp>
+
+#include <alpaka/queue/cpu/IGenericThreadsQueue.hpp>
+#include <alpaka/queue/QueueGenericThreadsNonBlocking.hpp>
+#include <alpaka/queue/QueueGenericThreadsBlocking.hpp>
+
+#include <openacc.h>
+
+namespace alpaka
+{
+    class DevOacc;
+
+    namespace traits
+    {
+        template<
+            typename TPltf,
+            typename TSfinae>
+        struct GetDevByIdx;
+    }
+    class PltfOacc;
+
+    namespace oacc
+    {
+        namespace detail
+        {
+            //#############################################################################
+            //! The OpenACC device implementation.
+            class DevOaccImpl
+            {
+            public:
+                //-----------------------------------------------------------------------------
+                DevOaccImpl(int iDevice) noexcept :
+                    m_deviceType(::acc_get_device_type()),
+                    m_iDevice(iDevice)
+                {}
+                //-----------------------------------------------------------------------------
+                DevOaccImpl(DevOaccImpl const &) = delete;
+                //-----------------------------------------------------------------------------
+                DevOaccImpl(DevOaccImpl &&) = delete;
+                //-----------------------------------------------------------------------------
+                auto operator=(DevOaccImpl const &) -> DevOaccImpl & = delete;
+                //-----------------------------------------------------------------------------
+                auto operator=(DevOaccImpl &&) -> DevOaccImpl & = delete;
+                //-----------------------------------------------------------------------------
+                ~DevOaccImpl() = default;
+
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST auto getAllExistingQueues() const
+                -> std::vector<std::shared_ptr<IGenericThreadsQueue<DevOacc>>>
+                {
+                    std::vector<std::shared_ptr<IGenericThreadsQueue<DevOacc>>> vspQueues;
+
+                    std::lock_guard<std::mutex> lk(m_Mutex);
+                    vspQueues.reserve(m_queues.size());
```

Comment:
```suggestion
                    std::lock_guard<std::mutex> lk(m_Mutex);
                    std::vector<std::shared_ptr<IGenericThreadsQueue<DevOacc>>> vspQueues(m_queues.size());
```

Reply:
`reserve()` is leaving the size of the vector 0, unlike `resize()`, which is what the ctor call amounts to.

Reply:
Oh sorry, my mistake. Good catch!